### PR TITLE
Run a server even quicker

### PR DESCRIPTION
**Pull request by pheyvaer**
Basic usage of server requires the existance of a configuration file. This might be tedious if you quickly want to run a server with one or more local files. In this case, you can use `instant-ldf-server`,
with at least one file path. For example, `instant-ldf-server data.nt` runs a server with a Turtle data source reading from the file `data.nt`. Note that the type of data source is detected based on the extension of the file. The supported extensions are `ttl`, `nt`, `hdt`, and `jsonld`.

- Is this something useful to include in the main repo? If not, I will create a separate package for that and just include the server as a library. 
- At the moment there is duplicate code in `ldf-server` and `instant-ldf-server`, if it's ok to add this functionality then I will clean that up first.

### CODE REVIEW

**Comment by pheyvaer** [comment]
I added 2 more examples. Are those the ones you were looking for?

**Comment by pheyvaer** [comment]
@thadguidry Is there an update needed before this can be merged? Besides cleaning up my code.

**Comment by rubensworks** [comment]
Perhaps @RubenVerborgh can have a look at this.

I see the value of this, but I'm a bit concerned about the amount of duplicate code between this and the other CLI script.

**Comment by pheyvaer** [comment]
Yes, I know of the duplicate code and will remove that if you would like the functionality to be added. So I will update the code before you do the actual merge :smile:

File: README.md [hunk 1]
```diff
@@ -197,6 +197,19 @@ $ docker run -p 3000:3000 -t -i --rm -v $(pwd)/config.json:/tmp/config.json ldf-
 
 You can [enable the Memento protocol](https://github.com/LinkedDataFragments/Server.js/wiki/Configuring-Memento) to offer different versions of an evolving dataset.
 
+### Run the server even faster
+
+Basic usage of server requires the existance of a configuration file.
```
Comment (RubenVerborgh) [inline README.md]: *existence

**Comment by RubenVerborgh** [comment]
Functionality is great!

Perhaps how this could be implemented without duplication, is that a config is automatically generated and passed to the other script?

Or it could be the same script with a `--source` or so parameter perhaps. No strong opinion.

**Comment by pheyvaer** [comment]
> Functionality is great!

:smile: 

> Perhaps how this could be implemented without duplication, is that a config is automatically generated and passed to the other script?

Indeed. Had that in mind too. Will have a look at it!

**Comment by pheyvaer** [comment]
I've removed the duplicate code. Let me know if it's fine or not. I see that one of the Travis builds fails, because of `hdt` for `Node.js:node`. I didn't change anything there and it works for versions 6 and 8. Anybody an idea what is going on there?

**Comment by rubensworks** [comment]
> I see that one of the Travis builds fails, because of hdt for Node.js:node. I didn't change anything there and it works for versions 6 and 8. Anybody an idea what is going on there?

Node-gyp doesn't work well yet on the latest Node version (12). So you can ignore this failure :-)

**Comment by mielvds** [comment]
All for this functionality! But yes, I also prefer the same cli with a --instant or --source or something

**Comment by pheyvaer** [comment]
OK, I can integrate it in the existing CLI, if most of you are in favour of that. @thadguidry @rubensworks Are you in favour of doing it that way?

**Comment by rubensworks** [comment]
Yep, something like `--source some-file.nt` for the existing CLI seems like a clean solution to me.

**Comment by pheyvaer** [comment]
In that case I would also (slightly) alter the current CLI. Instead of `server config.json [port [workers]]` you would get `server -c config.json [-p port] [-w workers]`. Else the use of `--source files` is inconsistent with the rest. Any objections?

**Comment by rubensworks** [comment]
Such a change would not be backwards-compatible, and would require a major version bump.
So either this PR would have to be merged into the dev branch then (which already contains breaking changes, but no concrete release ETA yet), or the CLI signature would need to be made backwards-compatible.

**Comment by pheyvaer** [comment]
Ok. Will keep the original one then.

**Comment by pheyvaer** [comment]
Updated. Usage is now
```
usage: server config.json [port [workers]]
       server -s|--sources [files]
```

**Comment by mielvds** [comment]
Can we get this merged?

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
   },
   "dependencies": {
     "asynciterator": "^1.1.0",
+    "commander": "^2.20.0",
```
Comment (rubensworks) [inline package.json]: This one doesn't seem necessary anymore.

Comment (mielvds) [reply package.json]: true. I anyway prefer minimist, which might be worth considering to keep the commandline a bit under control.

Comment (pheyvaer) [reply package.json]: Fixed.

**Comment by rubensworks** [comment]
@pheyvaer GitHub auto-closed this due to our branching changes. Feel free to re-open this PR. But it might require some changes, as the 3.0.0 release changes quite a bit of things.

**Pull request closed without merging**